commit f676c2338a318c537f256486f331faada38090b0
Author: Thomas Latham <T.Latham@warwick.ac.uk>
Date:   Tue Apr 20 14:34:10 2021 +0100

    Fix compilation with Pythia 8.304
    
    Summary: Pythia 8.304 switches to shared pointers (from raw) for ParticleDataEntry returns. Introduce version-dependent typedef to deal with this.
    
    Test Plan: Compiled successfully with 8.304, 8.303, 8.244
    
    Reviewers: jback, kreps
    
    Reviewed By: kreps
    
    Maniphest Tasks: T144
    
    Differential Revision: https://phab.hepforge.org/D68

diff --git a/src/EvtGenExternal/EvtPythiaEngine.cpp b/src/EvtGenExternal/EvtPythiaEngine.cpp
index df38ec8..72aa46d 100644
--- a/src/EvtGenExternal/EvtPythiaEngine.cpp
+++ b/src/EvtGenExternal/EvtPythiaEngine.cpp
@@ -38,6 +38,12 @@
 #include <iostream>
 #include <sstream>
 
+#if PYTHIA_VERSION_INTEGER < 8304
+typedef Pythia8::ParticleDataEntry* ParticleDataEntryPtr;
+#else
+typedef Pythia8::ParticleDataEntryPtr ParticleDataEntryPtr;
+#endif
+
 using std::endl;
 
 EvtPythiaEngine::EvtPythiaEngine( std::string xmlDir, bool convertPhysCodes,
@@ -430,9 +436,9 @@ void EvtPythiaEngine::updateParticleLists()
         // or assignment will give it a different memory address and it will no longer refer to
         // the original particleData information from the generator pointer.
 
-        Pythia8::ParticleDataEntry* entry_generic =
+        ParticleDataEntryPtr entry_generic =
             _genericPythiaGen->particleData.particleDataEntryPtr( PDGCode );
-        Pythia8::ParticleDataEntry* entry_alias =
+        ParticleDataEntryPtr entry_alias =
             _aliasPythiaGen->particleData.particleDataEntryPtr( PDGCode );
 
         // Check that the PDG code is not zero/null and exclude other
